qt/scientific_interfaces/EnggDiffraction/EnggDiffractionPresenter.h

```diff
@@ -238,15 +237,20 @@ protected slots:
                           std::string specNos);
 
   // algorithms to save the generated workspace
-  void saveGSS(std::string inputWorkspace, std::string bank, std::string runNo);
-  void saveFocusedXYE(std::string inputWorkspace, std::string bank,
-                      std::string runNo);
-  void saveOpenGenie(std::string inputWorkspace, std::string bank,
-                     std::string runNo);
+  void saveGSS(const std::string &inputWorkspace, const std::string &bank,
+               const std::string &runNo);
+  void saveFocusedXYE(const std::string &inputWorkspace,
+                      const std::string &bank, const std::string &runNo);
+  void saveNexus(const std::string &inputWorkspace, const std::string &bank,
+                 const std::string &runNumber);
+  void saveOpenGenie(const std::string &inputWorkspace, const std::string &bank,
+                     const std::string &runNo);
```

Comment:
If you're finding yourself having to pass the same info around a lot (i.e. theses three params `inputWorkspace`, `bank`, `runNumber`), do you think it would be nicer to use a `struct` to create a new concept? `FocusingOutputData` or something (think of a better name)?

Reply:
I agree it's nicer to replace `bank` and `runNumber` with `RunLabel`, which already exists all over the GUI. I've decided to leave `inputWorkspace` on its own as, to me at least, it makes more sense to keep the data and its 'filename' separate